### PR TITLE
fix: user is not asked to enable analytics after login - WPB-11332

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11332" title="WPB-11332" target="_blank">BugWPB-11332</a>  [iOS] Analytics user consent is not requested after login
  </td></table>
  <br />
 

### Issue

After logging in, the user should be asked if they wish to enable analytics. This functionality was removed recently (a regression). In this PR, we reintroduce the request, which is now based on the presence of the user preference:

- Previously the preference `disableAnalyticsSharing` was given a default value of `true`. Now we don't set a default, and instead consider a `nil` value as both `true` (analytics is disabled) and that the user hasn't yet made a choice.
- After login, if the user hasn't yet made a choice, then they will be prompted to enable analytics.

### Testing

After a clean install (with an analytics capable build) and login, you should be prompted. Further logins should not trigger a prompt.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ struct AnalyticsServiceConfigurationBuilder {
         return AnalyticsServiceConfiguration(
             secretKey: secretKey,
             serverHost: countlyURL,
-            didUserGiveTrackingConsent: !ExtensionSettings.shared.disableAnalyticsSharing
+            didUserGiveTrackingConsent: !(ExtensionSettings.shared.disableAnalyticsSharing ?? true)
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift:36]: maybe we should avoid to rely on the fallback value, for example making the isAnalyticsDisabled static this would be possible

```suggestion
            didUserGiveTrackingConsent: !TrackingManager.isAnalyticsDisabled
```

File: wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift [hunk 2]
```diff
@@ -236,6 +236,17 @@ final class ZClientViewController: UIViewController {
...
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        Task {
+            do {
+                try await trackingManager?.firstTimeRequestToEnableAnalytics()
+            } catch {
+                WireLogger.analytics.error("failed to first time enable analytics: \(error)")
+            }
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift:250]: suggestion move to own function

**Review by netbe — APPROVED** [review]
looks good just added a comment

File: wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift (hunk 2 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/UserInterface/MainController/ZClientViewController.swift:250]: done 6c8319105cd2a191a7e9c139b403a72b48082c2a

File: wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift (hunk 1 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/AnalytitcsServiceConfigurationBuilder.swift:36]: I'd like to avoid a static accessor, Swift concurrency won't like it. There's some strange dependency relationships here that we need to solve (but I would defer it for later): TrackingManager and the ExtensionSettings should really go to WireDomain where they can be used more directly.

File: wire-ios/Wire-iOS Tests/ExtensionSettingsTests.swift [hunk 3]
```diff
@@ -37,18 +37,14 @@ final class ExtensionSettingsTests: XCTestCase {
...
-        XCTAssertTrue(settings.disableAnalyticsSharing)
+        XCTAssertNil(settings.disableAnalyticsSharing)
 
         settings.disableAnalyticsSharing = false
-        XCTAssertFalse(settings.disableAnalyticsSharing)
+        XCTAssertEqual(settings.disableAnalyticsSharing, false)
 
         settings.disableAnalyticsSharing = true
-        XCTAssertTrue(settings.disableAnalyticsSharing)
+        XCTAssertEqual(settings.disableAnalyticsSharing, true)
```
Comment (samwyndham) [inline wire-ios/Wire-iOS Tests/ExtensionSettingsTests.swift:47]: Why did you change this? Just a style change?

File: wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift [hunk 4]
```diff
@@ -226,20 +230,37 @@ final class SettingsPropertyFactory {
...
 
             let setAction: SetAction = { [unowned self] _, value, resultHandler in
-                if var tracking = self.tracking {
-                    switch value {
-                    case .number(let number):
-                        tracking.disableAnalyticsSharing(isDisabled: number.boolValue, resultHandler: resultHandler)
-                    default:
-                        throw SettingsPropertyError.WrongValue("Incorrect type \(value) for key \(propertyName)")
+                guard let tracking else {
+                    return
```
Comment (samwyndham) [inline wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift:241]: Is this a valid state where `tracking` is nil?

**Review by samwyndham — APPROVED** [review]
Looks good. Added a couple of questions

File: wire-ios/Wire-iOS Tests/ExtensionSettingsTests.swift (hunk 3 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS Tests/ExtensionSettingsTests.swift:47]: Because it became optional

File: wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift (hunk 4 above)
Comment (johnxnguyen) [reply wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift:241]: It's hard to say. The tracking manager is passed around as optional in a lot of places, my guess it because for testing a nil value is passed in. I would like to replace the settings system with something much much simpler, which is why I didn't worry too much about making this code nice.

File: wire-ios/Wire-iOS Tests/ExtensionSettingsTests.swift (hunk 3 above)
Comment (samwyndham) [reply wire-ios/Wire-iOS Tests/ExtensionSettingsTests.swift:47]: 🤦 of course!

File: wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift (hunk 4 above)
Comment (samwyndham) [reply wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift:241]: 👍 Optionally consider logging an error if it is not there.
